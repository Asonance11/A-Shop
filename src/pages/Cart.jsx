import React from 'react';
import CartItem from '../components/CartItem';

const Cart = ({ cart, setCart }) => {
	const items = cart.map((item, index) => (
		<CartItem item={item} cart={cart} setCart={setCart} key={index + 100} />
	));

	return (
		<main className="pt-24 bg-darkGrey px-8 min-h-screen">
			<h2 className="text-3xl text-white font-bold font-serif italic text-center">
				Your Cart
			</h2>

			<section>{items}</section>
		</main>
	);
};

export default Cart;
