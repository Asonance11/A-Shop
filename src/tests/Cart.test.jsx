import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { describe, it, vi } from 'vitest';
import Cart from '../pages/Cart';

describe('Cart component', () => {
	it('renders correctly', async () => {
		const setCart = vi.fn();

		const { container } = render(<Cart cart={[]} setCart={setCart} />, {
			wrapper: MemoryRouter,
		});

		expect(container).toMatchSnapshot();
	});

	it('calls setCart when "checkout"  is clicked', async () => {
		const setCart = vi.fn();
		const user = userEvent.setup();

		render(<Cart cart={[]} setCart={setCart} />, { wrapper: MemoryRouter });

		const checkoutBtn = screen.getByRole('button', { name: 'Checkout' });

		await user.click(checkoutBtn);

		expect(setCart).toHaveBeenCalled();
	});
});
