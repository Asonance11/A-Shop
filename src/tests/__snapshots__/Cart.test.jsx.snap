// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cart component > renders correctly 1`] = `
<div>
  <main
    class="pt-24 bg-darkGrey px-8 min-h-screen"
  >
    <h2
      class="text-3xl text-white font-bold font-serif italic text-center"
    >
      Your Cart
    </h2>
    <section
      class="flex flex-col gap-4 mt-4"
    />
    <section
      class="flex items-center justify-between mt-2 text-white"
    >
      <p>
        Total Price:
      </p>
      <h3>
        $
        0
      </h3>
    </section>
    <button
      class="mt-4 py-2 px-4 bg-gradient-to-r from-forestGreen via-adventurousYellow to-oceanTurquoise rounded-md text-darkGrey transition-all duration-500 ease-in-out   hover:from-foggyBlue hover:to-forestGreen"
    >
      Checkout
    </button>
  </main>
</div>
`;
